src/components/typography/Heading/Heading.js

```diff
@@ -2,13 +2,16 @@ import PropTypes from 'prop-types';
 import React from 'react';
 
 import { headingSizes, headingColors, headingElements } from './Heading.styles';
+import spacerSizes from '../../../base/spacing';
 
-export const Heading = ({ children, color, level, size, ...others }) => {
+export const Heading = ({ children, color, level, size, margin, padding, ...others }) => {
```

Comment:
You can set default props for variables in a functional component like so:

```js
export const Heading = ({
  children,
  color = 'hot-pink',
  level = 1,
  size = 'big',
  margin,
  padding,
  …others
}) => {
```

That's actually just regular ES6/JavaScript so you can do it with any function.

Reply:
Oh yeah, I've been setting defaults for all/most properties. What I'm trying to figure out is how to have *different* defaults for different headings—so if it's a large heading, maybe the default margin is a bit bigger.

This may become less important when the Heading component is split into three or four different components, so if it's super tricky let's just leave things as is for now.

Reply:
You can't have conditional defaults based on _other_ defaults, so yeah—that sounds like you need multiple components. Or you'd have to define the defaults inside the component, but that seems messy. Separate components is better.